### PR TITLE
Add a script for reindexing the bags backlog

### PR DESCRIPTION
**Pull request by kenoir**
Part of https://github.com/wellcomecollection/platform/issues/4521

### CODE REVIEW

File: indexer/reindex_bags_backlog.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        },
+        'version': version,
+        'knownReplicas': {
+            'location': {
+                'provider': {
+                    "type": "AmazonS3StorageProvider"
+                },
+                'prefix': {
+                    "namespace": "go3dmZxK",
+                    "path": "XrLuVFbL"
```
Comment (alexwlchan) [inline indexer/reindex_bags_backlog.py]: Maybe worth setting these to explicit strings to make it easier to debug if it ends up in the wrong place?
```suggestion
                    "namespace": f"sent by {__file__}",
                    "path": f"sent by {__file__}"
```

File: indexer/reindex_bags_backlog.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        version = int(item['version']['N'])
+        stored_version = bags.get(dynamo_id, -1)
+
+        seen_bags = seen_bags + 1
+
+        if(seen_bags % 10 == 0 or seen_bags > (total_bags - 10)):
+            print(f"{seen_bags}/{total_bags}")
+
+        if(version > stored_version):
+            bags[dynamo_id] = version
```
Comment (alexwlchan) [inline indexer/reindex_bags_backlog.py]: You can use `tqdm` here if you want:

```suggestion
    for item in tqdm.tqdm(scan_table(dynamodb_client, TableName=table_name), total=total_bags):
        dynamo_id = item['id']['S']
        version = int(item['version']['N'])
        stored_version = bags.get(dynamo_id, -1)

        if version > stored_version:
            bags[dynamo_id] = version
```

File: indexer/reindex_bags_backlog.py [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+def publish_bags(sns_client, topic_arn, bags):
+    published_bags = []
+    unique_bags = len(bags)
+
+    for (dynamo_id, version) in bags.items():
+        space, external_id = dynamo_id.split("/", 1)
+        published_bags.append(dynamo_id)
+
+        payload = fake_known_replicas_payload(space, external_id, version)
+        #publish_payload(sns_client, topic_arn,  payload)
```
Comment (alexwlchan) [inline indexer/reindex_bags_backlog.py]: Why is this line comments out?

File: indexer/reindex_bags_backlog.py [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+        publish_count = len(published_bags)
+
+        if(publish_count % 10 == 0 or publish_count > (unique_bags - 10)):
+            print(f"{publish_count}/{unique_bags}")
```
Comment (alexwlchan) [inline indexer/reindex_bags_backlog.py]: Consider tqdm again?

File: indexer/reindex_bags_backlog.py [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+        'hostname': 'storage_bags_reindex_script/es_hostname',
+    }
+
+    dynamodb_client = create_client("dynamodb", role_arn)
+    sns_client = create_client("sns", role_arn)
+    elastic_client = create_elastic_client(role_arn, es_secrets)
+
+    bags_to_publish = get_latest_bags(dynamodb_client, table_name)
+    published_bags = publish_bags(sns_client, topic_arn, bags_to_publish)
+    not_indexed = confirm_indexed(elastic_client, published_bags)
```
Comment (alexwlchan) [inline indexer/reindex_bags_backlog.py:182]: Should the "confirm indexed" stuff live in a different script? Publishing everything to SNS feels like it'll be faster than getting it indexed.

If you do leave it here, include some context for the number you're printing below.

Comment (kenoir) [reply indexer/reindex_bags_backlog.py]: Going to make this use click and split commands that way.

File: indexer/reindex_bags_backlog.py (hunk 4 above)
Comment (kenoir) [reply indexer/reindex_bags_backlog.py]: Good idea, developed this in jupyter - where tqdm make sad.

Comment (alexwlchan) [reply indexer/reindex_bags_backlog.py]: <https://twitter.com/alexwlchan/status/1253607628841848834>:

> TIL: `from tqdm.notebook import tqdm` gets you a much nicer progress bar inside a Jupyter notebook than using regular tqdm!
>
>

File: indexer/reindex_bags_backlog.py (hunk 3 above)
Comment (kenoir) [reply indexer/reindex_bags_backlog.py]: Needs a proper dry_run flag.

**Pull request merged**